### PR TITLE
Add AOT compatibility checks documentation

### PR DESCRIPTION
**Pull request by m-redding**
After talking this over with @maririos, we decided that this documentation should be in the .NET repository, rather than the Azure SDK engineering hub website. The reasoning behind this is the following: 
1. Any content on the internal wiki page that is intended for both Azure SDK developers and partner teams is being migrated to engineering hub. So, this document should not be in the internal wiki.
2. This document is a little too specific to be included in the engineering hub website. It only applies to .NET, and is only useful to a small number of SDKs at the current point in time.

I added a link to its future location in the PowerShell script so it's easy to locate when debugging errors.

I also included a couple typo fixes I saw while looking at the docs here, and fixed a sentence I found unclear.

### CODE REVIEW

File: doc/dev/AotRegressionChecks.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    # [... other inputs]
+    CheckAOTCompat: true
+    AOTTestInputs:
+    - ArtifactName: Azure.Core
+      ExpectedAOTWarningsFilepath: /Azure.Core/tests/aotcompatibility/ExpectedAotWarnings.txt
+    - ArtifactName: Azure.Core.Experimental # For illustration only
+      ExpectedAOTWarningsFilepath: None
+```
+
+## How to resolve trimming warnings
```
Comment (jsquire) [inline doc/dev/AotRegressionChecks.md:63]:
```suggestion
## How to resolve trimming warnings

```

File: doc/dev/Using-Mock-Test-Generation.md [hunk 2]
```diff
@@ -137,7 +137,7 @@ Listening https on port: 8445
 
 # Execute Mock Test
 ## Trust the mock-service-host certificate '.ssh/localhost-ca.crt'
```
Comment (jsquire) [inline doc/dev/Using-Mock-Test-Generation.md:139]:
```suggestion

## Trust the mock-service-host certificate '.ssh/localhost-ca.crt'
```

**Review by jsquire — APPROVED** [review]

File: doc/dev/AotRegressionChecks.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# Enable AOT compatibility regression testing in CI pipelines
+An increasing number of .NET Azure SDK libraries are committed to making their libraries compatible with native AOT. To help support this work, there is now an opt-in pipeline step called "Check for AOT compatibility regressions in \[Package Name\]". This pipeline creates a small sample app that uses a project reference to collect the set of trimming warnings reported for the specified library. This approach for collecting warnings is described in [this article](https://learn.microsoft.com/dotnet/core/deploying/trimming/prepare-libraries-for-trimming?pivots=dotnet-8-0#show-all-warnings-with-test-app).
+
+## How to enable the pipeline for your package
+#### Collect any expected trimming warnings
```
[the remark this answers is not in the document]
Comment (m-redding) [reply doc/dev/AotRegressionChecks.md]: fixed!

File: doc/dev/AotRegressionChecks.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
+# Enable AOT compatibility regression testing in CI pipelines
+An increasing number of .NET Azure SDK libraries are committed to making their libraries compatible with native AOT. To help support this work, there is now an opt-in pipeline step called "Check for AOT compatibility regressions in \[Package Name\]". This pipeline creates a small sample app that uses a project reference to collect the set of trimming warnings reported for the specified library. This approach for collecting warnings is described in [this article](https://learn.microsoft.com/dotnet/core/deploying/trimming/prepare-libraries-for-trimming?pivots=dotnet-8-0#show-all-warnings-with-test-app).
```
[the remark this answers is not in the document]
Comment (m-redding) [reply doc/dev/AotRegressionChecks.md]: Added a link to the AOT overview

**Pull request merged**